include/jwt-cpp/jwt.h

```diff
@@ -2699,130 +2707,133 @@ namespace jwt {
-		using jwks_t = jwks<json_traits>;
+		using jwk_t = jwk<json_traits>;
 
 	public:
 
-		JWT_CLAIM_EXPLICIT jwks_keys(const typename json_traits::string_type& jwks_keys_token) {
+		JWT_CLAIM_EXPLICIT jwks(const typename json_traits::string_type& jwks_token) {
 			auto parse_claims = [](const typename json_traits::string_type& str) {
-				std::unordered_map<typename json_traits::string_type, jwks_t> res;
+				std::unordered_map<typename json_traits::string_type, jwk_t> res;
 				typename json_traits::value_type val;
 				if (!json_traits::parse(val, str))
 					throw std::runtime_error("Invalid json");
 
 				for (const auto& k : json_traits::as_object(val)) {
+					size_t id = 0;
 					for (const typename json_traits::value_type& item : json_traits::as_array(k.second)) {
-						jwks_t jwks_entry(item);
-						res.emplace(jwks_entry.get_key_id(), std::move(jwks_entry));
+						jwk_t jwk_entry(item);
+						if (jwk_entry.has_key_id())
+							res.emplace(jwk_entry.get_key_id(), std::move(jwk_entry));
+						else
+							res.emplace(std::to_string(id), std::move(jwk_entry));
+						++id;
 					}
 				}
 
 				return res;
 			};
 
-			jwks_keys_claims = parse_claims(jwks_keys_token);
+			jwks_claims = parse_claims(jwks_token);
 		}
 
 		/**
-		 * Get jwks
-		 * \return Requested jwks by key_id
-		 * \throw std::runtime_error If jwks was not present
+		 * Get jwk
+		 * \return Requested jwk by key_id
+		 * \throw std::runtime_error If jwk was not present
 		 */
-		jwks_t get_jwks(const typename json_traits::string_type& key_id) const {
-			if (!has_jwks(key_id)) {
-				typename json_traits::string_type error_msg = "jwks with key id \"";
+		jwk_t get_jwk(const typename json_traits::string_type& key_id) const {
+			if (!has_jwk(key_id)) {
+				typename json_traits::string_type error_msg = "jwk with key id \"";
 				error_msg += key_id;
 				error_msg += "\" not found";
 				throw std::runtime_error(error_msg.c_str());
 			}
 
-			return jwks_keys_claims.at(key_id);
+			return jwks_claims.at(key_id);
 		}
 
 		/**
 		 * Check if a jwks with the kid is present
 		 * \return true if jwks was present, false otherwise
 		 */
-		bool has_jwks(const typename json_traits::string_type& key_id) const noexcept { return jwks_keys_claims.count(key_id) != 0; }
+		bool has_jwk(const typename json_traits::string_type& key_id) const noexcept { return jwks_claims.count(key_id) != 0; }
 
-		bool empty() const noexcept { return jwks_keys_claims.empty(); }
+		bool empty() const noexcept { return jwks_claims.empty(); }
+		
+		std::vector<jwk_t> to_vector() {
+			std::vector<jwk_t> v_jwks_claims;
+			v_jwks_claims.reserve(jwks_claims.size());
+			for (const auto& element : jwks_claims)
+   				v_jwks_claims.push_back(element.second);
+			return v_jwks_claims;
+		}
 
 	private:
 
-		std::unordered_map<typename json_traits::string_type, jwks_t> jwks_keys_claims;
+		std::unordered_map<typename json_traits::string_type, jwk_t> jwks_claims;
```

Comment:
Only because i've implemented this and used a few different auth servers... I am really unsure about this being a map...

What if there are two keys neither have KIDs? One would be lost, most likely the new key that is starting to issue tokens.